irksome/imex.py

```diff
@@ -506,11 +519,27 @@ def advance(self):
             for ghatbit, kbit in zip(ghat.subfunctions, ks[i].subfunctions):
                 ghatbit += dtc * AA[i, i] * kbit
 
-            chat.assign(C_hat[i+1])
-            self.mass_solver.solve()
-            self.num_mass_nonlinear_iterations += self.mass_solver.snes.getIterationNumber()
-            self.num_mass_linear_iterations += self.mass_solver.snes.getLinearSolveIterations()
-            k_hat_s[i + 1].assign(khat)
+        self._finalize()
```

Comment:
This seems reasonable.  Do the four IMEX/DIRK tableaux you've implemented actually hit all of these cases in the test suite?

Reply:
I could go fishing for more methods, but I don't actually know of one that would allow `_finalize_no_last_explicit` and doesn't satisfy the conditions for `_finalize_stiffly_accurate`.  None of the ones in the Ascher-Ruuth-Spiteri paper would do so.

I tested `_finalize_no_last_explicit` locally, just by hacking the code to not call the other routine.  Another option would just be a specialized test that overrides `self._finalize` after calling `__init__` in order to be sure we hit that code.

Reply:
If there aren't any methods that actually go that route, we're probably fine.

Reply:
Yes - presumably if we add new methods that exercise the code, we'll also add new tests...